File: get_extrema_min.py
#!/usr/bin/python3

import os
import time
from datetime import date
from datetime import datetime
from datetime import timedelta
import shutil
from ftplib import FTP

test = False
#difaxMinDir = '/home/disk/data/archive/images/difax/difax_min'
difaxMinDir = '/home/disk/data/images/difax/difax_min'
tempDir = '/tmp'
catalogPrefix = 'surface.NWS'
    
# Field Catalog inputs
if test:
    ftpCatalogServer = 'ftp.atmos.washington.edu'
    ftpCatalogUser = 'anonymous'
    ftpCatalogPassword = 'brodzik@uw.edu'
    catalogDestDir = 'brodzik/incoming/impacts'
else:
    ftpCatalogServer = 'catalog.eol.ucar.edu'
    ftpCatalogUser = 'anonymous'
    catalogDestDir = '/pub/incoming/catalog/impacts'

# Open ftp connection to NCAR sever
if test:
    catalogFTP = FTP(ftpCatalogServer,ftpCatalogUser,ftpCatalogPassword)
    catalogFTP.cwd(catalogDestDir)
else:
    catalogFTP = FTP(ftpCatalogServer,ftpCatalogUser)
    catalogFTP.cwd(catalogDestDir)

# Look for yesterday's map
today = datetime.utcnow()
yesterday = today - timedelta(days = 1)
dateStr = yesterday.strftime("%Y%m%d")
#yearMonthStr = yesterday.strftime("%Y%m")

# note for 2021: new maps are in DifaxMinDir+'/'+year+month
# e.g. /home/disk/data/archive/images/difax/difax_min/202112
#for file in os.listdir(os.path.join(difaxMinDir,yearMonthStr)):
for file in os.listdir(difaxMinDir):
    if dateStr in file and file.endswith('gif'):
        print('file = ',file)
        (dateTime,ext) = file.split('.')
        date = dateTime[0:8]
        print('date = ',date)
        catalogName = catalogPrefix+'.'+dateTime+'.min_temp.gif'
        print('catalogName = ',catalogName)
    
        #shutil.copy(os.path.join(difaxMinDir,yearMonthStr,file),
	#	    tempDir+'/'+catalogName)
        shutil.copy(os.path.join(difaxMinDir,file),
		    tempDir+'/'+catalogName)
            
        # ftp file to catalog location
        ftpFile = open(os.path.join(tempDir,catalogName),'rb')
        catalogFTP.storbinary('STOR '+catalogName,ftpFile)
        ftpFile.close()
        
        # remove file from tempDir
        os.remove(os.path.join(tempDir,catalogName))

# Look for today's map
dateStr = today.strftime("%Y%m%d")
#yearMonthStr = today.strftime("%Y%m")

#for file in os.listdir(os.path.join(difaxMinDir,yearMonthStr)):
for file in os.listdir(difaxMinDir):
    if dateStr in file and file.endswith('gif'):
        print('file = ',file)
        (dateTime,ext) = file.split('.')
        date = dateTime[0:8]
        print('date = ',date)
        catalogName = catalogPrefix+'.'+dateTime+'.min_temp.gif'
        print('catalogName = ',catalogName)
    
        #shutil.copy(os.path.join(difaxMinDir,month,file),
	#	    tempDir+'/'+catalogName)
        shutil.copy(os.path.join(difaxMinDir,file),
		    tempDir+'/'+catalogName)
            
        # ftp file to catalog location
        ftpFile = open(os.path.join(tempDir,catalogName),'rb')
        catalogFTP.storbinary('STOR '+catalogName,ftpFile)
        ftpFile.close()

        # remove file from tempDir
        os.remove(os.path.join(tempDir,catalogName))

# Close ftp connection
catalogFTP.quit()

File: 2020_VERSIONS/get.extrema_min.pl
#!/usr/bin/perl

$DIFAX_MIN_PATH = "/home/disk/data/archive/images/difax/difax_min";

$ARCHIVE_PATH = "/home/disk/funnel/impacts/archive/ops/extrema";

# Look for yesterday's map

$month = `date --date='yesterday' "+%Y%m"`;
chop $month;
#print "month = $month\n";
$yesterday = `date --date='yesterday' "+%Y%m%d"`;
chop $yesterday;
#print "yesterday = $yesterday\n";
$yesterdayDir = $ARCHIVE_PATH."/".$yesterday;
unless (-e $yesterdayDir) {
    mkdir $yesterdayDir;
}

# note for 2021: new maps are in $DIFAX_MIN_PATH."/".$year.$month
# e.g. /home/disk/data/archive/images/difax/difax_min/202008
chdir($DIFAX_MIN_PATH."/".$month);
#print "changed to $DIFAX_MIN_PATH/$month\n";
foreach $file (<$yesterday*1200.gif>) {
  print "file = $file\n";
  ($dateTime) = ($file =~ /(.*)\.gif/);
  $date = substr($dateTime,0,8);
  #print "date = $date\n";
  $outFile = $ARCHIVE_PATH."/".$date."/ops.extrema.".$dateTime.".min_temp.gif";
  print "outFile = $outFile\n";
  unless (-e $outFile) {
    system("/bin/cp $file $outFile");
  }
}

# Look for today's map

$month = `date "+%Y%m"`;
chop $month;
#print "month = $month\n";
$today = `date "+%Y%m%d"`;
chop $today;
#print "today = $today\n";
$todayDir = $ARCHIVE_PATH."/".$today;
unless (-e $todayDir) {
    mkdir $todayDir;
}

chdir($DIFAX_MIN_PATH."/".$month);
#print "changed to $DIFAX_MIN_PATH/$month\n";
foreach $file (<$today*1200.gif>) {
  print "file = $file\n";
  ($dateTime) = ($file =~ /(.*)\.gif/);
  $date = substr($dateTime,0,8);
  #print "date = $date\n";
  $outFile = $ARCHIVE_PATH."/".$date."/ops.extrema.".$dateTime.".min_temp.gif";
  print "outFile = $outFile\n";
  unless (-e $outFile) {
    system("/bin/cp $file $outFile");
  }
}

exit(0);
